source/D2CMP/src/Count.cpp

```diff
@@ -1 +1,155 @@
-#include "Count.h" 
+/**
+ * D2MOO
+ * Copyright (c) 2020-2023  The Phrozen Keep community
+ *
+ * This file belongs to D2MOO.
+ * File authors: Mir Drualga
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "Count.h"
+
+#include <cstdlib>
+
+#include "Fog.h"
+
+/**
+ * 1.00: D2CMP.0x100085D0
+ * 1.07 Beta: D2CMPd.0x6FC7DA50
+ * 1.07: Inline or unused
+ * 1.10: Inline or unused
+ *
+ * This implementation outputs effectively the same binary from 1.00.
+ */
+int __fastcall CountConsecutiveFirst(const BYTE* pSrc, const BYTE* pEnd, int nMaxCount, BOOL* pbCountEqualsDist, BOOL* pbIsRemainingSame)
```

Comment:
Could this be implemented in terms of `CountConsecutive` ?

Reply:
Yes, but if aiming for binary output similarity, these two functions are copy-pasted, but exist as two separate functions.

If binary output similarity is not a priority, I can implement this in terms of `CountConsecutive`.

Reply:
I don't think we actually want exact binary outputs for D2Moo, we already reworked a lot of code using C++ features that were not available for VC6.
But if it's important for you we can merge this as is.

Reply:
Sounds good to me.